src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -297,8 +295,12 @@ class LicenceService(
       request.additionalConditions.transformToEntityAdditional(licenceEntity, request.conditionType)
 
     val newAdditionalConditions =
-      submittedAdditionalConditions.filter { licenceEntity.additionalConditions.none { submittedCondition -> submittedCondition.conditionCode == it.conditionCode } }
-        .map { newAdditionalCondition -> newAdditionalCondition.copy(expandedConditionText = newAdditionalCondition.conditionText, licence = licenceEntity) }
+      submittedAdditionalConditions
+        .filter { licenceEntity.additionalConditions.none { submittedCondition -> submittedCondition.conditionCode == it.conditionCode } }
+        .map { it.copy(expandedConditionText = it.conditionText, licence = licenceEntity) }
+        // Persist conditions to allow creating back reference in new data items
+        .map { additionalConditionRepository.saveAndFlush(it) }
```

Comment:
I think this is fine. It would be more efficient to save here, then flush after the function chain is complete to persist in one DB query. It could cause issues without trying it, and the performance gains would be very minimal all things considered.